src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -450,13 +429,12 @@ private sealed class ScriptEntryPoint : SynthesizedEntryPointSymbol
             //     var script = new Script();
             //     script.<Initialize>().GetAwaiter().GetResult();
             // }
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
             {
-                // CreateBody should only be called if no errors.
-                Debug.Assert((object)_getAwaiterMethod != null);
-                Debug.Assert((object)_getResultMethod != null);
-
                 var syntax = DummySyntax();
+                var compilation = _containingType.DeclaringCompilation;
+                var binder = compilation.GetBinder((CSharpSyntaxNode)compilation.SyntaxTrees[0].GetRoot());
```

Comment:
@cston: I'm pretty sure that the containing type is also synthesized, so it wouldn't have an associated tree